src/datasets/tasks/base.py

```diff
@@ -0,0 +1,26 @@
+import abc
+from dataclasses import dataclass
+from typing import Dict
+
+from ..features import Features
+
+
+@dataclass
```

Comment:
I'm not sure about this being a `dataclass`

I'd rather have `task`, `input_schema` and `label_schema` be class attributes, so they can be shared by all instances of the `TaskTemplate` concrete class.

On the other hand, for the classification task, we need the label_featrues to be instance specific 🤔

Reply:
Actually we can use dataclasses AND class attributes together, if we type hint the class attributes with `ClassVar`:

```python
from typing import ClassVar

# [...]

@dataclass
class TaskTemplate(abc.ABC):
    task: ClassVar[str]
    input_schema: ClassVar[Features]
# [...]
```

The attributes marked as `ClassVar` will be ignored by the "dataclass mechanism" (left out of `__init__`, `__eq__`, etc)

Source: https://docs.python.org/3/library/dataclasses.html#class-variables

Reply:
i did not know about `ClassVar` so thank you for the suggestion and pointed to the docs! i've now added this